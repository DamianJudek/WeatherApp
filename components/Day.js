import React from 'react';
import { View, Text, StyleSheet, Image } from 'react-native';
import PropTypes from 'prop-types';
import SlideInView from './SlideInView';

const styles = StyleSheet.create({
  icon: {
    height: 55,
    width: 55,
  },
  floatRight: {
    alignSelf: 'flex-end',
  },
  floatleft: {
    alignSelf: 'flex-start',
  },
  baseText: {
    fontFamily: 'Roboto',
    color: '#fff',
    fontSize: 18,
    textAlign: 'center',
  },
  smallText: {
    fontFamily: 'Roboto',
    color: '#fff',
    fontSize: 14,
    textAlign: 'center',
  },
  partition: {
    width: '100%',
    height: 0,
    borderTopWidth: 1,
    borderColor: 'rgba(255, 255, 255, 0.2)',
    transform: [
      {
        rotate: '-45deg',
      },
    ],
  },
  block: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
  },
});

const Day = ({ weather, style }) => (
  <SlideInView delay={900} style={style}>
    <Text style={styles.baseText}>{weather.name}</Text>

    <View style={styles.block}>
      <Image
        style={[styles.icon, styles.floatleft]}
        resizeMode="contain"
        source={{
          uri: `http://openweathermap.org/img/wn/${weather.iconDay}@2x.png`,
        }}
      />
      <Text
        style={[
          styles.smallText,
          styles.floatleft,
        ]}>{`${weather.maxTemp}\u2103`}</Text>
      <View style={styles.partition}></View>
      <Text
        style={[
          styles.smallText,
          styles.floatRight,
        ]}>{`${weather.minTemp}\u2103`}</Text>
      <View>
        <Image
          style={[styles.icon, styles.floatRight]}
          resizeMode="contain"
          source={{
            uri: `http://openweathermap.org/img/wn/${weather.iconNight}@2x.png`,
          }}
        />
      </View>
    </View>
  </SlideInView>
);

Day.propTypes = {
  weather: PropTypes.object,
  style: PropTypes.oneOfType([PropTypes.array, PropTypes.object]),
};

export default Day;
