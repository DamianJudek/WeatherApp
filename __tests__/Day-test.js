import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import Day from '../components/Day';

jest.mock('react-native/Libraries/Animated/src/NativeAnimatedHelper');
test('renders correctly', () => {
  const snap = renderer
    .create(
      <Day
        style={{
          borderColor: '#fff',
          borderRightWidth: 1,
          flex: 1,
          justifyContent: 'center',
          paddingHorizontal: '2%',
        }}
        weather={{
          iconDay: '10d',
          iconNight: '04n',
          maxTemp: 18,
          minTemp: 11,
          name: 'SOB.',
        }}
      />,
    )
    .toJSON();
  expect(snap).toMatchSnapshot();
});
