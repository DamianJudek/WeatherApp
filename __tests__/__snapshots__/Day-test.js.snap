// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "borderColor": "#fff",
      "borderRightWidth": 1,
      "flex": 1,
      "justifyContent": "center",
      "opacity": 0,
      "paddingHorizontal": "2%",
      "transform": Array [
        Object {
          "translateY": 35,
        },
      ],
    }
  }
>
  <Text
    style={
      Object {
        "color": "#fff",
        "fontFamily": "Roboto",
        "fontSize": 18,
        "textAlign": "center",
      }
    }
  >
    SOB.
  </Text>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <Image
      resizeMode="contain"
      source={
        Object {
          "uri": "http://openweathermap.org/img/wn/10d@2x.png",
        }
      }
      style={
        Array [
          Object {
            "height": 55,
            "width": 55,
          },
          Object {
            "alignSelf": "flex-start",
          },
        ]
      }
    />
    <Text
      style={
        Array [
          Object {
            "color": "#fff",
            "fontFamily": "Roboto",
            "fontSize": 14,
            "textAlign": "center",
          },
          Object {
            "alignSelf": "flex-start",
          },
        ]
      }
    >
      18℃
    </Text>
    <View
      style={
        Object {
          "borderColor": "rgba(255, 255, 255, 0.2)",
          "borderTopWidth": 1,
          "height": 0,
          "transform": Array [
            Object {
              "rotate": "-45deg",
            },
          ],
          "width": "100%",
        }
      }
    />
    <Text
      style={
        Array [
          Object {
            "color": "#fff",
            "fontFamily": "Roboto",
            "fontSize": 14,
            "textAlign": "center",
          },
          Object {
            "alignSelf": "flex-end",
          },
        ]
      }
    >
      11℃
    </Text>
    <View>
      <Image
        resizeMode="contain"
        source={
          Object {
            "uri": "http://openweathermap.org/img/wn/04n@2x.png",
          }
        }
        style={
          Array [
            Object {
              "height": 55,
              "width": 55,
            },
            Object {
              "alignSelf": "flex-end",
            },
          ]
        }
      />
    </View>
  </View>
</View>
`;
